trie/stacktrie.go

```diff
@@ -169,7 +196,7 @@ func (t *StackTrie) insert(st *stNode, key, value []byte, path []byte) {
 
 		// Add new child
 		if st.children[idx] == nil {
-			st.children[idx] = newLeaf(key[1:], value)
+			st.children[idx] = newLeaf(key[1:], value, t.preHashedLeafs)
```

Comment:
So we wouldn't pass `false` here because it's a `branchNode`, and only pass `t.preHashedLeafs` in the `leafNode` case?